### PR TITLE
eig extension

### PR DESCRIPTION
**Pull request by zhumakhan**
#6180

### CODE REVIEW

File: ivy/functional/experimental/linear_algebra.py [hunk 1]
```diff
@@ -123,3 +123,67 @@ def kron(
...
+
+
+@to_native_arrays_and_back
+@handle_nestable
+@handle_exceptions
+def eig(
+    x: Union[ivy.Array, ivy.NativeArray],
+    /,
+) -> Tuple[ivy.Array, ...]:
+    """Computes eigenvalies and eigenvectors of x. Returns an tuple with two elements:
```
Comment (Nightcrab) [inline ivy/functional/experimental/linear_algebra.py:135]: typo

File: ivy/array/experimental/linear_algebra.py [hunk 2]
```diff
@@ -60,3 +60,24 @@ def kron(
...
+    ) -> Tuple[ivy.Array, ...]:
+        """
+        ivy.Array instance method variant of ivy.eig.
+        This method simply wraps the function, and so the docstring for
+        ivy.eig also applies to this method with minimal changes.
+
+        Examples
+        --------
+         >>> x = ivy.array([[1,2], [3,4]])
+        >>> ivy.eig(x)
```
Comment (Nightcrab) [inline ivy/array/experimental/linear_algebra.py:76]: should be ``x.eig()``

File: ivy/container/experimental/linear_algebra.py [hunk 3]
```diff
@@ -170,3 +170,106 @@ def kron(
...
+        x: Union[ivy.Array, ivy.NativeArray, ivy.Container],
+        /,
+        *,
+        key_chains: Optional[Union[List[str], Dict[str, str]]] = None,
+        to_apply: bool = True,
+        prune_unapplied: bool = False,
+        map_sequences: bool = False,
+    ) -> ivy.Container:
+        """
+        ivy.Container instance method variant of ivy.eig.
```
Comment (Nightcrab) [inline ivy/container/experimental/linear_algebra.py:185]: this is a static method not instance method

File: ivy/container/experimental/linear_algebra.py [hunk 4]
```diff
@@ -170,3 +170,106 @@ def kron(
...
+        Parameters
+        ----------
+            x
+                container with input arrays.
+
+        Returns
+        -------
+            ret
+                container including arrays corresponding
+                eigenvealues and eigenvectors of input arrays
```
Comment (Nightcrab) [inline ivy/container/experimental/linear_algebra.py:198]: typo

File: ivy/container/experimental/linear_algebra.py [hunk 5]
```diff
@@ -170,3 +170,106 @@ def kron(
...
+
+        Examples
+        --------
+        >>>c=ivy.Container({'x':{'xx':x}})
+        >>>ivy.eig(c)
```
Comment (Nightcrab) [inline ivy/container/experimental/linear_algebra.py:203]: should be ``ivy.Container.eig``

File: ivy/container/experimental/linear_algebra.py [hunk 6]
```diff
@@ -170,3 +170,106 @@ def kron(
...
+
+        Returns
+        -------
+            ret
+                container including arrays corresponding
+                eigenvealues and eigenvectors of input arrays
+
+        Examples
+        --------
+        >>>c=ivy.Container({'x':{'xx':x}})
```
Comment (Nightcrab) [inline ivy/container/experimental/linear_algebra.py:202]: x should be defined in the example

File: ivy/container/experimental/linear_algebra.py [hunk 7]
```diff
@@ -170,3 +170,106 @@ def kron(
...
+        }
+        >>>ivy.eig(c)['x']['xx']
+        (
+            ivy.array([-0.37228107+0.j,  5.3722816 +0.j]),
+            ivy.array([
+                    [-0.8245648 +0.j, -0.41597357+0.j],
+                    [0.56576747+0.j, -0.9093767 +0.j]
+                ])
+        )
+        """
```
Comment (Nightcrab) [inline ivy/container/experimental/linear_algebra.py:268]: examples should be ``c.eig()``, and ``x`` should be defined

File: ivy/functional/backends/torch/experimental/linear_algebra.py [hunk 8]
```diff
@@ -100,3 +100,9 @@ def kron(
 
 
 kron.support_native_out = True
+
+
+def eig(x: torch.Tensor, /) -> Tuple[torch.Tensor, ...]:
+    if not torch.is_complex(x):
+        return torch.linalg.eig(x.to(torch.complex128))
+    return torch.linalg.eig(x)
```
Comment (Nightcrab) [inline ivy/functional/backends/torch/experimental/linear_algebra.py:108]: torch ein returns a namedtuple not a tuple, this needs to be handled.
```
import ivy

ivy.set_backend("torch")
x = ivy.array([[1,2], [3,4]])

print(ivy.eig(x))
"""
torch.return_types.linalg_eig(
    eigenvalues=ivy.array([-0.37228132+0.j,  5.37228132+0.j]),
    eigenvectors=ivy.array([[ 0.82456484+0.j,  0.41597356+0.j],
        [-0.56576746+0.j,  0.90937671+0.j]])
)
"""

ivy.set_backend("jax")
x = ivy.array([[1,2], [3,4]])

print(ivy.eig(x))
"""
(ivy.array([-0.37228107+0.j,  5.3722816 +0.j]), ivy.array([[-0.8245648 +0.j, -0.41597357+0.j],
       [ 0.56576747+0.j, -0.9093767 +0.j]]))
"""```

**Review by Nightcrab — CHANGES_REQUESTED** [review]

**Comment by zhumakhan** [comment]
Hi @Nightcrab, thanks a lot for thorough comments, Requested changes are made,

**Pull request merged**